pcap/pcap.go

```diff
@@ -455,6 +456,7 @@ func (p *Handle) compileBPFFilter(expr string) (_Ctype_struct_bpf_program, error
 // CompileBPFFilter compiles and returns a BPF filter for the pcap handle.
 func (p *Handle) CompileBPFFilter(expr string) ([]BPFInstruction, error) {
 	bpf, err := p.compileBPFFilter(expr)
+	defer C.pcap_freecode(&bpf)
```

Comment:
I think this one will fail, since compileBPFFilter may already call freecode (new line 449), and this would make it freecode twice on the same object.